orangecontrib/text/widgets/owcollocations.py

```diff
@@ -0,0 +1,191 @@
+from collections import namedtuple
+import numpy as np
+
+from Orange.data.util import get_unique_names
+from Orange.widgets.data.owrank import TableModel, TableView
```

Comment:
It is not nice to import classes from widgets since they can be changed dramatically without previous deprecations (we do not assume that classes inside the widget file are used anywhere else). 

Since there is not so much code also used in this widget, I suggest copying TableModela and TableView from the rank widget and removing unused code.

Reply:
I agree. I think both classes can be used elsewhere (what does Extract Keywords use?). I can prepare the PR and clean this one.

Reply:
@PrimozGodec Is this what you had in mind? https://github.com/biolab/orange3/pull/6098

Reply:
I had in mind to take the implementation from the rank and copy it into the Collocations widget. However, your proposed solution is excellent if other widgets use the same implementation (maybe Score Documents and Keywords can).

The only problem is that we would need to wait for Orange's release before we can merge this PR.

Reply:
Right. I can do either. I think having a nice table capable of showing certain scores as colored bars might be useful. @janezd @markotoplak @VesnaT What do you think?